pkg/engine/plugins.go

```diff
@@ -56,6 +56,33 @@ func (p pluginSet) Union(other pluginSet) pluginSet {
 	return newSet
 }
 
+// Removes less specific entries.
+//
+// For example, the plugin aws would be removed if there was an already existing plugin
+// aws-5.4.0.
+func (p pluginSet) Deduplicate() pluginSet {
+	existing := map[string]workspace.PluginInfo{}
+	newSet := newPluginSet()
+	add := func(p workspace.PluginInfo) {
+		prev, ok := existing[p.Name]
+		if ok && p.Version == nil && p.PluginDownloadURL == "" {
+			// no new information
+			return
+		}
+
+		if prev.Version == nil && prev.PluginDownloadURL == "" {
```

Comment:
Little confusing that this line works because prev will be a zero value struct when ok is false. Maybe both this if conditions should be wrapped in an "if ok {" block to be clear that we're only doing comparisons if there's something to compare to.